core/blockchain.go

```diff
@@ -1924,11 +1932,18 @@ func (bc *BlockChain) processBlock(block *types.Block, statedb *state.StateDB, s
 	ptime := time.Since(pstart)
 
 	vstart := time.Now()
-	if err := bc.validator.ValidateState(block, statedb, receipts, usedGas); err != nil {
+	if err := bc.validator.ValidateState(block, statedb, receipts, usedGas, false); err != nil {
 		bc.reportBlock(block, receipts, err)
 		return nil, err
 	}
 	vtime := time.Since(vstart)
+
+	if witness := statedb.Witness(); witness != nil {
+		if err = bc.validator.ValidateWitness(witness, block.ReceiptHash(), block.Root()); err != nil {
+			bc.reportBlock(block, receipts, err)
```

Comment:
Notably, this method will persist the bad block. 

It should work fine in normal cases. However, what I concern, if one client for cross-validating is buggy and the correct block is tagged as bad due to the fault. We need to make sure the chain can actually recover eventually.

 We just need to make sure the bad blocks won't prevent chain recovering.

Reply:
I'll actually do a different cross validation than this. I'm moving it over to the engine API. Will have this removed in the final production code, it's just testing for now.